transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/DependencyConfiguration.java

```diff
@@ -5,24 +5,125 @@
  */
 package com.linkedin.transport.plugin;
 
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
 /**
  * Represents a dependency to be applied to a certain sourceset configuration (e.g. implementation, compileOnly, etc.)
  * In the future can expand to incorporate exclude rules, dependency substitutions, etc.
  */
 public class DependencyConfiguration {
   private ConfigurationType _configurationType;
-  private String _dependencyString;
+  private String _module;
+  private String _version;
+  private String _classifier;
+  private Set<PackageModule> _excludedPackageModules;
```

Comment:
If we go by the recommendation that we just pass the `Set` of `Map` here, then this should be called `_excludeProperties`.